javascript/ql/src/semmle/javascript/security/dataflow/FileAccessToHttp.qll

```diff
@@ -0,0 +1,87 @@
+/** 
+ * Provides Taint tracking configuration for reasoning about file access taint flow to html post body 
+ */
+import javascript
+import semmle.javascript.frameworks.HTTP
+
+module FileAccessToHttpDataFlow {
+  /**
+   * A data flow source for code injection vulnerabilities.
+   */
+  abstract class Source extends DataFlow::Node { }
+
+  /**
+   * A data flow sink for code injection vulnerabilities.
+   */
+  abstract class Sink extends DataFlow::Node { }
+
+  /**
+   * A sanitizer for code injection vulnerabilities.
+   */
+  abstract class Sanitizer extends DataFlow::Node { }
+
+  /**
+   * A taint-tracking configuration for reasoning about code injection vulnerabilities.
+   */
+  class Configuration extends TaintTracking::Configuration {
+    Configuration() { this = "FileAccessToHttpDataFlow" }
+
+    override predicate isSource(DataFlow::Node source) {
+      source instanceof Source
+    }
+
+    override predicate isSink(DataFlow::Node sink) {
+      sink instanceof Sink
+    }
+
+    override predicate isSanitizer(DataFlow::Node node) {
+      super.isSanitizer(node) or
+      node instanceof Sanitizer
+    }
+  }
+
+/** Taint step propagating a tainted parameter or a  variable to a wrapper object. */ 
+  private class ObjectTaintStep extends TaintTracking::AdditionalTaintStep {
```

Comment:
Adding this taint step can lead to significant imprecision: if `v` is tainted and you have an assignment `o.p = v`, then `o` will become tainted, so any (!) read of a property (not just `p`) from `o` will also be considered tainted. Do you have a concrete use case in mind that you want to model?

Reply:
I'll also weigh in here that `AdditionalTaintStep` should not be subclassed in a query-specific file; there is a performance penalty when doing so. That class is only for taint steps that are shared by all taint configurations.

Steps that are specific to a configuration should be added in `isAdditionalTaintStep` (even if it's the only configuration used by the query).